File: recyclerview-diff/src/main/java/cn/ads/recyclerview/CommonDiffItemCallback.java
package cn.ads.recyclerview;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import java.util.Objects;

/**
 * @author QiuJie
 */
public class CommonDiffItemCallback extends DiffUtil.ItemCallback<Object> {
    @Override
    public boolean areItemsTheSame(@NonNull Object oldItem, @NonNull Object newItem) {
        return oldItem == newItem;
    }

    @Override
    public boolean areContentsTheSame(@NonNull Object oldItem, @NonNull Object newItem) {
        return Objects.equals(oldItem, newItem);
    }
}

File: ads/src/main/kotlin/ads/view/recyclerview/CommonDiffItemCallback.kt
package ads.view.recyclerview

import androidx.recyclerview.widget.DiffUtil
import java.util.*

/**
 * @author QiuJie
 */
class CommonDiffItemCallback : DiffUtil.ItemCallback<Any>() {
    override fun areItemsTheSame(oldItem: Any, newItem: Any): Boolean {
        return oldItem === newItem
    }

    override fun areContentsTheSame(oldItem: Any, newItem: Any): Boolean {
        return Objects.equals(oldItem, newItem)
    }
}